### PR TITLE
[Feature](Job Schedule)implement Transient Task Register

### PR DESCRIPTION
**Pull request by BePPPower**
## Proposed changes

Issue Number: close #xxx

1. Implement the `TransientTaskRegister` to support submitting transient tasks which do not require a timer trigger.
2. rename some class:
    `TimerTaskDisruptor` -> `TaskDisruptor`
    `TimerTaskEvent` -> `TaskEvent`
    `TimerTaskExpirationHandler` -> `TaskHandler`
    `AsyncJobManager` -> `TimerJobManager`
    `MemoryTask` -> `TransientTask`

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.04 seconds
 stream load tsv:          512 seconds loaded 74807831229 Bytes, about 139 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          68 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.7 seconds inserted 10000000 Rows, about 336K ops/s
 storage size: 17162284025 Bytes

**Comment by BePPPower** [comment]
run feut

**Comment by BePPPower** [comment]
run p0

**Comment by BePPPower** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/scheduler/disruptor/TaskHandler.java [hunk 1]
```diff
@@ -119,12 +134,27 @@ public void onEventTask(TimerTaskEvent timerTaskEvent) {
         jobTaskManager.addJobTask(jobTask);
     }
 
+    public void onMemoryTask(TaskEvent taskEvent) {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/scheduler/disruptor/TaskHandler.java:137]:
```suggestion
    public void onTransientTaskSubmit(TaskEvent taskEvent) {
```

File: fe/fe-core/src/main/java/org/apache/doris/scheduler/disruptor/TaskType.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package org.apache.doris.scheduler.disruptor;
+
+public enum TaskType {
+    TimerJobTask,
+
+    MemoryTask
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/scheduler/disruptor/TaskType.java:23]:
```suggestion
    TransientTask
```

File: fe/fe-core/src/main/java/org/apache/doris/scheduler/job/DorisTimerTask.java [hunk 3]
```diff
@@ -27,8 +27,8 @@
 
 /**
  * This class represents a timer task that can be scheduled by a Netty timer.
- * When the timer task is triggered, it produces an event task using the Disruptor.
- * The event task contains the ID of the event and the ID of the task itself.
+ * When the timer task is triggered, it produces a Job task using the Disruptor.
+ * The Job task contains the ID of the Job and the ID of the task itself.
  */
 @Getter
 public class DorisTimerTask implements TimerTask {
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/scheduler/job/DorisTimerTask.java:34]: Better not name a class with `Doris` name

**Comment by BePPPower** [comment]
run buildall external

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.1 seconds
 stream load tsv:          544 seconds loaded 74807831229 Bytes, about 131 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.4 seconds inserted 10000000 Rows, about 340K ops/s
 storage size: 17162082059 Bytes

**Comment by BePPPower** [comment]
run feut

**Review by CalvinKirs — APPROVED** [review]

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**